src/main/java/com/arpnetworking/clusteraggregator/aggregation/AggMessageExtractor.java

```diff
@@ -52,6 +53,18 @@ public String entityId(final Object message) {
                     .append(metricData.getMetric())
                     .append("||")
                     .append(metricData.getPeriod());
+            for (final Messages.DimensionEntry dimensionEntry : metricData.getDimensionsList()) {
+                final String k = dimensionEntry.getKey();
+                if (!(k.equals(AggClientConnection.CLUSTER_KEY)
```

Comment:
Wait - why do you think its a map? [messages.proto](https://github.com/ArpNetworking/metrics-aggregator-protocol/blob/master/src/main/proto/messages.proto) just uses a list of `String, String` pairs, right? Or does protobuf have some special way of knowing the first element of `DimensionEntry` should be unique?